haystack/fields.py

```diff
@@ -134,7 +135,10 @@ def prepare_template(self, obj):
             template_names = ['search/indexes/%s/%s_%s.txt' % (app_label, model_name, self.instance_name)]
 
         t = loader.select_template(template_names)
-        return t.render(Context({'object': obj}))
+        if django.VERSION >= (1, 8, 0):
+            return t.render({'object': obj})
```

Comment:
Should probably use `Context({'object': obj}).flatten()` instead (right now it's missing default constants like None, True, False).

for example:
```python
ctx = Context({'object': obj})
if django.VERSION >= (1, 8, 0):
    ctx = ctx.flatten()
t.render(ctx)
```